### PR TITLE
basic configuration for paywall

### PR DESCRIPTION
**Pull request by cellog**
# Description

This pulls in the basic configuration for the paywall to make it run. It does not have any of the testing or deploy configuration yet.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [X] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: paywall/next.config.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const withSourceMaps = require('@zeit/next-source-maps')
+
+const copyFile = promisify(fs.copyFile)
+
+module.exports = withSourceMaps({
+  publicRuntimeConfig: {
+    unlockEnv: process.env.UNLOCK_ENV || 'dev',
+    httpProvider: process.env.HTTP_PROVIDER || '127.0.0.1',
+    readOnlyProvider: process.env.READ_ONLY_PROVIDER,
+    locksmithHost: process.env.LOCKSMITH_HOST,
```
Comment (akeem) [inline paywall/next.config.js]: I don't think will be needed for the paywall, not show stopping and can be removed later

File: paywall/next.config.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    return config
+  },
+  exportPathMap: async (defaultPathMap, { dev, dir, outDir }) => {
+    // Export robots.txt and humans.txt in non-dev environments
+    if (!dev && outDir) {
+      await copyFile(
+        join(dir, '..', 'unlock-app', 'src', 'static', 'robots.txt'),
+        join(outDir, 'robots.txt')
```
Comment (akeem) [inline paywall/next.config.js]: I think this can be pruned

File: paywall/next.config.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      )
+      await copyFile(
+        join(dir, '..', 'unlock-app', 'src', 'static', 'humans.txt'),
+        join(outDir, 'humans.txt')
```
Comment (akeem) [inline paywall/next.config.js]: We can probably chop this as well

File: paywall/next.config.js (hunk 1 above)
Comment (cellog) [reply paywall/next.config.js]: the locksmithHost yes? I agree. This is mainly about getting the existing code over, there will be a lot of pruning once things are here.

File: paywall/next.config.js (hunk 2 above)
Comment (cellog) [reply paywall/next.config.js]: I think we will want robots.txt, nobody should crawl the paywall

File: paywall/next.config.js (hunk 3 above)
Comment (cellog) [reply paywall/next.config.js]: I agree, actually, because this will display in the console for publisher's websites

**Comment by cellog** [comment]
committed based on your feedback, PTAL

**Pull request merged**